lib/test_track_rails_client/assignment_helper.rb

```diff
@@ -1,19 +1,39 @@
 module TestTrackRailsClient::AssignmentHelper
-  def stub_test_track_assignments(assignment_registry) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def stub_test_track_assignments(assignment_registry) # rubocop:disable Metrics/AbcSize
     raise "Cannot stub test track assignments when TestTrack is enabled" if TestTrack.enabled?
 
-    split_registry = {}
-    assignments = []
+    schema = TestTrackRailsClient::Schema.new
 
+    assignments = []
     assignment_registry.each do |split_name, variant|
-      assignment_registry[split_name] = variant.to_s
-      split_registry[split_name] = { variant => 100 }
+      raise "Variant #{variant} for #{split_name} must be a string or symbol" unless [String, Symbol].include?(variant.class)
+      raise "Split #{split_name} does not exist in #{schema.file_path}" unless schema.splits.include?(split_name)
+
       assignments << { split_name: split_name.to_s, variant: variant.to_s, unsynced: false }
     end
 
     visitor_attributes = { id: "fake_visitor_id", assignments: assignments }
 
     allow(TestTrack::Remote::Visitor).to receive(:fake_instance_attributes).and_return(visitor_attributes)
+  end
+
+  def stub_test_track_split_registry # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    raise "Cannot stub test track split registry when TestTrack is enabled" if TestTrack.enabled?
+
+    schema = TestTrackRailsClient::Schema.new
+
+    split_registry = {}
+    schema.splits.each do |split_name, weighting_registry|
+      default_variant = weighting_registry.keys.sort.first
+
+      adjusted_weightings = { default_variant => 100 }
+      weighting_registry.except(default_variant).keys.each do |variant|
+        adjusted_weightings[variant] = 0
+      end
+
+      split_registry[split_name] = adjusted_weightings
+    end
+
```

Comment:
So, I think going along with what John was saying, we may be able to push this code down into the place where we are currently doing the fuzzing as a replacement approach. Then we will also get this behavior in development if someone isn't relying on a real TT server.